mark.c

```diff
@@ -637,7 +643,7 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
     /* The following is 0 only for small objects described by a simple  */
     /* length descriptor.  For many applications this is the common     */
     /* case, so we try to detect it quickly.                            */
-    if (descr & ((~(WORDS_TO_BYTES(SPLIT_RANGE_WORDS) - 1)) | GC_DS_TAGS)) {
+    if (descr & ((~(PTRS_TO_BYTES(SPLIT_RANGE_WORDS) - 1)) | GC_DS_TAGS)) {
```

Comment:
Hmm, so now I'm wondering if we should override `WORDS_TO_BYTES` iff `RISCV && CHERI`?

Reply:
fixed in commit [408f667b](https://github.com/capablevms/bdwgc/pull/1/commits/408f667bb59d4d389e44ff70d5631bf922413ed9)